src/compiler/nvcc.rs

```diff
@@ -401,25 +407,45 @@ pub fn generate_compile_commands(
         .unwrap()
         .path;
 
+    // Build nvcc's internal files in `$TMPDIR/$hash_key` so the paths are
+    // stable across compilations. This is important because this path ends
+    // up in the preprocessed output, so using random tmpdir paths leads to
+    // erroneous cache misses.
+    let out_dir = env::temp_dir().join("sccache_nvcc").join({
+        // Combine `hash_key` with the output path in case
+        // the same file is concurrently built to separate
+        // output paths.
+        let mut m = crate::util::Digest::new();
+        m.update(hash_key.as_bytes());
+        m.update(cwd.join(output).as_os_str().as_encoded_bytes());
+        m.finish()
+    });
+    fs::create_dir_all(&out_dir).ok();
```

Comment:
`let _ = fs::create_dir_all(..)` would be preferable, although I don't understand how it'd be acceptable to fail creating the directory?

Reply:
That's fine with me

Reply:
Done